fastapi/openapi/utils.py

```diff
@@ -190,12 +190,32 @@ def get_openapi_path(
                         response.setdefault("content", {}).setdefault(
                             "application/json", {}
                         )["schema"] = response_schema
-                    status_text = http.client.responses.get(int(additional_status_code))
+                    try:
+                        # check if valid HTTP status code
+                        status_text: Optional[str] = http.client.responses.get(
+                            int(additional_status_code)
+                        )
+                    except ValueError as ex:
+                        status_groups: Dict[str, str] = {
```

Comment:
Probably better to put the status groups dict at the module level (near `validation_error_definition`)

Then this can be much more concise:
```
status_text: Optional[str] = status_groups.get(str(additional_status_code).lower())
status_text = status_text or http.client.responses.get(int(additional_status_code))
```

Reply:
Indeed it reads more clearly than the try-except block